ext/src/ruby_api/execution_error.rs

```diff
@@ -39,6 +41,13 @@ impl ExecutionError {
         self.path.clone()
     }
 
+    pub fn to_h(&self) -> RHash {
+        let ruby_h = rhash_with_capacity(3);
+        let _ = ruby_h.aset("message", self.message());
```

Comment:
This I wasn't quite sure of -- the compiler told me I should use the `Result` that came from here, but I don't have any idea what kind of error I might expect. Should I handle this differently?

Reply:
A lot of `magnus` methods will return `Result<something, magnus::Error>`, and `RHash::aset` is one of those. I don't know all of the cases when that would happen, but I think one would be trying to modify a frozen hash. There are a couple options here:
1. `unwrap` all of the errors because we don't expect them to happen and we're okay with panicking if an error were to ever be returned.
2. Have the return type of the method be `Result<RHash, Error>`, which when exposed to Ruby will surface any error as an exception. Then, when calling `aset` we can use `?` syntax ([docs](https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-question-mark-operator)) to return the error or continue in the `Ok` case, like:
```rust
ruby_h.aset("message", self.message)?;
...
Ok(ruby_h)
```
I lean towards option 2 but I don't have strong feelings either way.

Also, optionally we could use [`bulk_insert`](https://docs.rs/magnus/0.5.4/magnus/r_hash/struct.RHash.html#method.bulk_insert) here, something like:
```rust
ruby_h.bulk_insert(&[("message", self.message), ("path", self.path)])?;
```

Reply:
I just read about `?` but didn't consider it here -- handy! Thanks for the links to magnus docs, I'll keep those open ....

Reply:
I ended up switching back to `aset` because it looks like `bulk_insert` needs you do to some prep work before it will use the given values as keys: 

```
error[E0277]: the trait bound `(&str, Option<Vec<std::string::String>>): ReprValue` is not satisfied
   --> ext/src/ruby_api/execution_error.rs:46:28
    |
46  |         ruby_h.bulk_insert(&[("path", self.path), ("message", self.message())])
    |                ----------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ReprValue` is not implemented for `(&str, Option<Vec<std::string::String>>)`
```

Also, I used `self.path()` instead of `self.path` because: 

```
error[E0507]: cannot move out of `self.path` which is behind a shared reference
  --> ext/src/ruby_api/execution_error.rs:46:29
   |
46 |         ruby_h.aset("path", self.path)?;
   |                             ^^^^^^^^^ move occurs because `self.path` has type `Option<Vec<std::string::String>>`, which does not implement the `Copy` trait
```

And similarly with `self.message()`:

```
error[E0277]: the trait bound `Cow<'_, str>: TypedData` is not satisfied
   --> ext/src/ruby_api/execution_error.rs:47:32
    |
47  |         ruby_h.aset("message", self.message);
    |                ----            ^^^^^^^^^^^^ the trait `TypedData` is not implemented for `Cow<'_, str>`
    |                |
    |                required by a bound introduced by this call
    |
```

(I don't understand those error messages yet but I could make them go away 😅 )